### PR TITLE
Update DG for Parser component and use cases

### PR DESCRIPTION
**Pull request by czhi-bin**
image

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -379,14 +371,84 @@ Use case ends.
...
-    Use case resumes at step 2.
+Use case ends.
+
+
+**Use case: Getting help manual**
+
+**MSS**
+
+1. User requests to for a help manual
+2. contHACKS displays the help manual
```
Comment (tanruiquan) [inline docs/DeveloperGuide.md:439]: I am not too sure if this is considered adding low level details but here's my suggestion

```suggestion
1. User requests to for a help manual
2. contHACKS displays the link to the help manual
3. User clicks on the link
4. contHACKS redirect users to the User Guide
```

Comment (James-Kua) [inline docs/DeveloperGuide.md:439]: Maybe we can do something like:
```suggestion
1. User requests to for a help manual
2. contHACKS redirects user to the User Guide in the browser
```

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -310,33 +310,21 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
 
 **MSS**
 
-1. User requests to create a contact
-2. contHACKS shows a list of existing tags in contHACKS, and asks for input regarding details of the contact
-3. User inputs details of the contact
-4. contHACKS reflects that contact has been created successfully
+1. User requests to create a contact with the inputted details
```
Comment (James-Kua) [inline docs/DeveloperGuide.md:313]: This sentence seems odd although grammatically correct. Maybe we can simplify it?
```suggestion
1. User requests to create a contact and inputs contact details
```

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -310,33 +310,21 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
-* 2a. There are no tags in contHACKS<br />
-    * 2a1. contHACKS reflects that there are no tags in contHACKS<br />
-      Use case ends.
-
-
-* 3a. User does not give a tag to the contact.
-    * 3a1. Contact creation continues without an input tag<br />
-    Use case resumes at step 4.
-
+* 1a. User gives insufficient details for contact creation
```
Comment (James-Kua) [inline docs/DeveloperGuide.md:319]:
```suggestion
* 1a. User fails to provide compulsory field.
```

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -348,28 +336,32 @@ Use case ends.
...
 
 **Use case: Updating a contact**
 
 **MSS**
 
-1. User requests to update a specific contact
-2. contHACKS returns the current details of the contact
-3. User inputs the new details of the contact
-4. contHACKS reflects that contact is successfully updated
+1. User requests to update a specific contact with the inputted details
```
Comment (James-Kua) [inline docs/DeveloperGuide.md:363]: Same suggestion and above

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (czhi-bin) [reply docs/DeveloperGuide.md:439]: I feel like this also contains UI details? Like the [example talking about good and bad](https://nus-cs2103-ay2122s1.github.io/website/se-book-adapted/chapters/specifyingRequirements.html#details-2)

Comment (driip-ddrop) [reply docs/DeveloperGuide.md:439]: I think don't need to include the displays link and the clicks on the link part.

File: docs/DeveloperGuide.md [hunk 5]
```diff
@@ -379,14 +371,84 @@ Use case ends.
...
+
+Use case ends.
+
+
+**Use case: Exiting the application**
+
+**MSS**
+
+1. User requests to exit the application
+2. contHACKS closed.
```
Comment (driip-ddrop) [inline docs/DeveloperGuide.md:449]: Not sure if this matters but I think shouldn't be in past tense.
```suggestion
2. contHACKS closes.
```

**Review by driip-ddrop — APPROVED** [review]
LGTM! But just wondering, are there supposed to be full stop at the end of each sentence? Cause some parts got full stop some don't have

**Pull request merged**